datalad/interface/ls.py

```diff
@@ -189,6 +196,89 @@ def annex_local_size(self):
         info = self.info
         return info['local annex size'] if info else None
 
+
+@auto_repr
+class FsModel(DsModel):
+
+    __slots__ = ['_path', '_info', '_repo']
+
+    def __init__(self, path, repo=""):
+        self._path = path  # can be overridden
+        self._info = None
+        self._repo = repo
+        self._branch = None
+
+    @property
+    def _symlink(self):
+        if islink(self._path):                    # if symlink
+            target_path = readlink(self._path)    # find link target
+            # convert to absolute path if not
+            target_path = opj(dirname(self._path), target_path) if not isabs(target_path) else target_path
+            return target_path if exists(target_path) else False
+        return False
+
+    @property
+    def repo(self):
+        if exists(opj(self._repo, ".git", "annex")):
+            return AnnexRepo(self._repo)
+        elif exists(opj(self._repo, ".git")):
+            return GitRepo(self._repo)
+        else:
+            return None
+
+    @property
+    def date(self):
+        """Date of last modification
+        """
+        if self._type is not ['git', 'annex']:
+            return lstat(self._path).st_mtime
+        else:
+            super(self.__class__, self).date
+
+    @property
+    def size(self):
+        _type = self._type
+        if not _type:
+            return -1
+        if 'annex' in _type:
+            return self.annex_local_size
+        elif 'git' in _type:
+            return self.git_local_size
+        elif 'file' in _type:
+            return lstat(self._path).st_size
+        elif 'broken-link' in _type:
+            return 0
+        elif 'link' in _type:
+            return lstat(self._symlink).st_size
+        elif 'dir' in _type:
+            return lstat(self._path).st_size  # add du -s command for plain dir
+        else:
+            return -1
+
+    @property
+    def _type(self):
+        if islink(self.path):
```

Comment:
Here we should also better ask AnnexRepo either file has content or not... we might publish to some crippled filesystem which would get annex into direct mode, so no symlinks would be there